src/medCore/process/medRunnableProcess.cpp

```diff
@@ -52,11 +52,17 @@ dtkAbstractProcess * medRunnableProcess::getProcess()
 
 void medRunnableProcess::internalRun()
 {
-    if (d->process) {
-        if (d->process->update() == 0)
+    if (d->process)
+    {
+        int res = d->process->update();
+
+        if (res == 0)
             emit success (this);
         else
+        {
             emit failure (this);
+            emit displayError(res);
```

Comment:
The problem here is that a toolbox cannot add it's own behaviour to the specific error, since it has no way of retrieving it. I agree with @LoicCadour that adding it to the ```failure ``` signal might be a good idea.